jobs/move_extract.py

```diff
@@ -33,62 +33,89 @@ def sort():
+            group, now_skew = mat.groups()
+            if group:
+                if tests:
                     problems.append('Duplicate %s' % job)
                     break
-                skew = args
+                tests = group
                 continue
+            if now_skew:
+                if skew:
+                    problems.append('Duplicate skew %s' % job)
+                skew = now_skew
         else:
-            if not skew:
-                continue
-            for arg in values['args']:
-                if not arg.startswith('--check_version_skew'):
-                    continue
-                if arg != skew:
-                    problems.append('Mismatch in %s: %s != %s' % (job, arg, skew))
+            new_args = []
+            stop = False
+            for arg in args:
+                these = None
+                add = True
+                if (
+                        arg == '--env-file=jobs/pull-kubernetes-federation-e2e-gce.env'
+                        and not job == 'pull-kubernetes-federation-e2e-gce'):
+                    these = r'--ginkgo.skip=\[Slow\]|\[Serial\]|\[Disruptive\]|\[Flaky\]|\[Feature:.+\]'  # pylint: disable=line-too-long
+                elif (
+                        arg == '--env-file=jobs/pull-kubernetes-e2e.env'
+                        and not job.startswith('pull-kubernetes-federation-e2e-gce')):
+                    these = r'--ginkgo.skip=\[Slow\]|\[Serial\]|\[Disruptive\]|\[Flaky\]|\[Feature:.+\]'  # pylint: disable=line-too-long
+                elif arg == '--env-file=jobs/suite/slow.env':
+                    these = r'--ginkgo.focus=\[Slow\] --ginkgo.skip=\[Serial\]|\[Disruptive\]|\[Flaky\]|\[Feature:.+\]'  # pylint: disable=line-too-long
+
+                elif arg == '--env-file=jobs/suite/serial.env':
+                    these = r'--ginkgo.focus=\[Serial\]|\[Disruptive\] --ginkgo.skip=\[Flaky\]|\[Feature:.+\]'  # pylint: disable=line-too-long
+                    add = False
+                elif arg == '--env-file=jobs/suite/default.env':
+                    these = r'--ginkgo.skip=\[Slow\]|\[Serial\]|\[Disruptive\]|\[Flaky\]|\[Feature:.+\]'  # pylint: disable=line-too-long
+                if add:
+                    new_args.append(arg)
+                if not these:
                     continue
-                break
-            else:
-                values['args'].append(skew)
+                if tests:
+                    problems.append('Duplicate end %s' % job)
+                    stop = True
+                    break
+                tests = these
+            if stop:
+                continue
+            args = new_args
+
+            testing = '--test=false' not in args
+
+            if not testing:
+                if skew:
+                    problems.append('Cannot skew kubectl without tests %s' % job)
+                if tests:
+                    problems.append('Cannot --test_args when --test=false %s' % job)
+                continue
```

Comment:
maybe check if `--test` is true but `--test_args` is None here, that seems should not happen.